### PR TITLE
Lower the probability of not passing "Transactions are propagated through a network".

### PR DESCRIPTION
**Pull request by Cifko**
## Description
There is still a chance of not passing this test. But now it's reduced from 1% to 0.01%.
The tests now requires all-1 (that means all but 1 node) to have the transaction in their mempool.

### CODE REVIEW

**Comment by hansieodendaal** [comment]
As proposed please consider changing the acceptance criteria to success if _round up to 1% of total nodes_ less than the total number of nodes received the transaction.

**Comment by Cifko** [comment]
> As proposed please consider changing the acceptance criteria to success if _round up to 1% of total nodes_ less than the total number of nodes received the transaction.

Done.
I still left the comment there. Because it can still happen (1 out of ~10k), so we will not spend time investigating again.

File: integration_tests/features/support/steps.js [hunk 1]
```diff
@@ -902,8 +907,13 @@ Then(
       );
       this.lastResult = await client.transactionState(sig);
```
Comment (hansieodendaal) [inline integration_tests/features/support/steps.js:908]: We will wait a really long time here for a node that did not receive the transaction; suggest making the wait time here for `async () => client.transactionStateResult(sig)` a lot shorter and evaluate a test condition to decide if we should wait again. Sample code below (_I could not figure out query the total seeds and nodes so do not mind the first bit_):

``` rust
    let succeeded = 0;
    let total = 0;
    for (const seed in this.seeds) {
      total += 1;
    }
    for (const seed in this.nodes) {
      total += 1;
    }
    canFail = canFail ? parseInt(canFail) : 0;
    const needed = Math.floor(total * (1 - canFail / 100));
    await this.forEachClientAsync(async (client, name) => {
      for (let i = 0; i < 1100; i++) {
        if (succeeded < needed) {
          await waitFor(
            async () => client.transactionStateResult(sig),
            pool,
            1 * 1000
          );
        }
        this.lastResult = await client.transactionState(sig);
        if (this.lastResult.result === pool) {
          succeeded += 1;
          break;
        }
      }
      console.log(
        `Node ${name} response is: ${this.lastResult.result}`,
        "  ",
        succeeded,
        "of",
        total,
        "(need",
        needed,
        ")"
      );
    });
```
Sample output when one node did not get the transaction:
```

File: integration_tests/features/support/steps.js [hunk 2]
```diff
@@ -902,8 +907,13 @@ Then(
...
       console.log(`Node ${name} response is: ${this.lastResult.result}`);
-      expect(this.lastResult.result).to.equal(pool);
+      if (this.lastResult.result === pool) succeeded += 1;
```
Comment (hansieodendaal) [inline integration_tests/features/support/steps.js]: We will wait a really long time here for a node that did not receive the transaction; suggest making the wait time here for `async () => client.transactionStateResult(sig)` a lot shorter and evaluate a test condition to decide if we should wait again. Sample code below (_I could not figure out how to query the total seeds and nodes so do not mind the first bit_):

``` rust
    let succeeded = 0;
    let total = 0;
    for (const seed in this.seeds) {
      total += 1;
    }
    for (const seed in this.nodes) {
      total += 1;
    }
    canFail = canFail ? parseInt(canFail) : 0;
    const needed = Math.floor(total * (1 - canFail / 100));
    await this.forEachClientAsync(async (client, name) => {
      for (let i = 0; i < 1100; i++) {
        if (succeeded < needed) {
          await waitFor(
            async () => client.transactionStateResult(sig),
            pool,
            1 * 1000
          );
        }
        this.lastResult = await client.transactionState(sig);
        if (this.lastResult.result === pool) {
          succeeded += 1;
          break;
        }
      }
      console.log(
        `Node ${name} response is: ${this.lastResult.result}`,
        "  ",
        succeeded,
        "of",
        total,
        "(need",
        needed,
        ")"
      );
    });
```
Sample output when one node did not get the transaction:
```
Node SENDER response is: MEMPOOL    1 of 17 (need 16 )
Node SeedNode7 response is: MEMPOOL    2 of 17 (need 16 )
Node SeedNode2 response is: MEMPOOL    3 of 17 (need 16 )
Node SeedNode5 response is: MEMPOOL    4 of 17 (need 16 )
Node SeedNode4 response is: MEMPOOL    5 of 17 (need 16 )
Node SeedNode0 response is: MEMPOOL    6 of 17 (need 16 )
Node SeedNode3 response is: MEMPOOL    7 of 17 (need 16 )
Node BaseNode6 response is: MEMPOOL    8 of 17 (need 16 )
Node BaseNode4 response is: MEMPOOL    9 of 17 (need 16 )
Node BaseNode2 response is: MEMPOOL    10 of 17 (need 16 )
Node BaseNode7 response is: MEMPOOL    11 of 17 (need 16 )
Node SeedNode6 response is: MEMPOOL    12 of 17 (need 16 )
Node BaseNode3 response is: MEMPOOL    13 of 17 (need 16 )
Node BaseNode5 response is: MEMPOOL    14 of 17 (need 16 )
Node BaseNode0 response is: MEMPOOL    15 of 17 (need 16 )
Node BaseNode1 response is: MEMPOOL    16 of 17 (need 16 )
Node SeedNode1 response is: NOT_STORED    16 of 17 (need 16 )
Stopping nodes
```

Comment (Cifko) [reply integration_tests/features/support/steps.js]: Ok, I've changed it from Promise.all in the _worlds.js_ to a custom promise that ends when failed count or success count is reached.

File: integration_tests/features/support/world.js [hunk 3]
```diff
@@ -188,16 +188,39 @@ class CustomWorld {
...
+            console.log(`${succeeded} of ${total} (need ${total - canFail})`);
+            if (succeeded >= total - canFail) resolve();
+          },
+          () => {
+            failed += 1;
+            if (failed > canFail) reject("Too many failed.");
+          }
+        );
+      }
+    });
```
Comment (hansieodendaal) [inline integration_tests/features/support/world.js:223]: Really nice way of doing this

**Review by hansieodendaal — APPROVED** [review]
LGTM

**Pull request merged**